src/ompl/tools/thunder/src/SPARSdb.cpp

```diff
@@ -959,11 +959,15 @@ bool ompl::geometric::SPARSdb::addStateToRoadmap(const base::PlannerTerminationC
 
     ++iterations_;
 
-    //@TODO - Ramy: Test if creating another nbhd to seperate recall from connectivity has positive effects.
-    findGraphNeighbors(qNew, graphNeighborhood, visibleNeighborhood);
-    if (denseRoadmap_)
+    if (denseRoadmap_) {
         findGraphNeighbors(qNew, gnbhd, vnbhd, granularity_);
+        if (vnbhd.size()) {
+            return false;
+        }
+    }
 
+    //@TODO - Ramy: Test if creating another nbhd to seperate recall from connectivity has positive effects.
+    findGraphNeighbors(qNew, graphNeighborhood, visibleNeighborhood);
```

Comment:
`findGraphNeighbors` can be called twice if (denseRoadmap_); sure that's what you want?

Reply:
Yes! The first call is on a very small neighborhood, it takes less than 1 ms. This is to determine whether or not to add the node. The second one is the expensive call (can take up to a minute), this is the one to determine which edges to add between the new node and nodes in the graph.

Reply:
got it. could you add this info as comment? would be useful to readers saying that it's intentional

Reply:
I added a comment implying it is intentional and clarifying that it is very cheap!